src/lib.rs

```diff
@@ -116,7 +119,19 @@ mod arch {
             command.arg(perf_output);
         }
 
+        // perf script can take a long time to run. Notify the user that it is running
+        // by using a spinner. Note that if this function exits before calling
+        // spinner.finish(), then the spinner will be completely removed from the terminal.
+        let spinner = ProgressBar::new_spinner().with_prefix("Running perf script");
+        spinner.set_style(
+            ProgressStyle::with_template("{prefix} [{elapsed}]: {spinner:.green}")
+                .unwrap()
+                .tick_strings(&[".", "..", "...", " ..", "  .", "", "✓"]),
+        );
+        spinner.enable_steady_tick(Duration::from_millis(500));
+
         let output = command.output().context("unable to call perf script")?;
+        spinner.finish();
```

Comment:
So we can more cleanly always call `spinner.finish()` after `command.output()` (before yielding the error using the `?` operator). I think I would prefer that.

Reply:
The behavior here will depend on what happens with our discussion down at https://github.com/flamegraph-rs/flamegraph/pull/343#discussion_r1835270990, so let me just lay out some details. I don't really feel strongly about any of this, but let's resolve that discussion first before we try to resolve this one. Here are some details about why I want to do these in that particular order:

I've been playing around with different scenarios to see what happens in different situations with the spinner. What I have found is that the resources are always cleaned up when the spinner goes out of scope. The only thing that calling `spinner.finish()` seems to do is leave a copy of the prefix/message/final tick in the terminal history. Note that the final tick string is configurable, and it is only called once finish() gets set (not part of the spinner animation), so using a special final tick symbol can act as an indicator of success.

If you don't call `spinner.finish()` before the spinner goes out of scope, then the line where the spinner was is effectively removed from the terminal history, as if the spinner wasn't ever there. If you call finish whether the command succeeds or fails, then we will see the terminal history including the final tick. The one exception is that if you don't use a message/prefix and the default tick is blank, then nothing will be included in the terminal history.

With all of this in mind, if you use the default spinner, the final tick of the spinner is simply blank, so it doesn't really indicate anything. My custom ticks use a checkmark to indicate successful completion. The effect of doing what you propose with the default spinner would be to always have the spinner line left in the terminal history regardless of whether the command succeeded or failed. I think that is OK. If we use a custom final tick, then I think this change would be confusing, because the user might think the command succeeded. If we actually hit the error case here, then in reality it means that the process was never started up in the first place, so I felt that just removing the spinner message from the terminal history made sense.

Reply:
Right, I still think it makes sense to unconditionally call `.finish()` so we leave the elapsed time in terminal history, which I think will be useful in both the success and error cases.